the-powder-toy: 97.0.352 -> 98.2.365

LordGrimmauld opened the pull request:
## Description of changes
Changelog: https://github.com/The-Powder-Toy/The-Powder-Toy/compare/v97.0.352...v98.2.365

Additional (nix) changes:
- moved package from pkgs/games to pkgs/by-name
  - for that purpose, restructured the parameter list to be backwards compatible with what was overrideable from all-packages.nix
- drop python dependency, it is just a dependency of meson and does not need to be specified explicitly
- make use of more modern nix tooling to install the desktop file
- add missing preInstall and postInstall run hooks
- drop unnecessary patches that were previously needed to compile on gcc13: the patch was merged upstream
- bump lua dependency from 5.1 to the current lua (5.2)
- ordered parameters and build inputs to be alphabetical
- ran nixpkgs-fmt

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

LordGrimmauld committed:
the-powder-toy: 97.0.352 -> 98.2.365

LordGrimmauld commented:
Note: I do not own any darwin machines, i am waiting for ofborg to build this but can't actually run the binaries either. Do *not* merge if darwin tests didn't succeed yet. And feedback from darwin users is welcome.

LordGrimmauld commented:
Ok, ofborg logs look fine. Quick heads-up by a darwin user would still be appreciated but it seems to work as intended.

eclairevoyant commented on pkgs/by-name/th/the-powder-toy/package.nix:
```diff
@@ -0,0 +1,75 @@
+, Cocoa ? darwin.apple_sdk.frameworks.Cocoa # all-packages.nix used to allow overriding Cocoa, this is fully backwards-compatible
+, copyDesktopItems
+, curl
+, darwin
+, fetchFromGitHub
+, fetchpatch
```
```suggestion
```

eclairevoyant replied:
```suggestion
, fetchFromGitHub
```

eclairevoyant commented on pkgs/by-name/th/the-powder-toy/package.nix:
```diff
@@ -0,0 +1,75 @@
+  src = fetchFromGitHub {
+    owner = "The-Powder-Toy";
+    repo = "The-Powder-Toy";
+    rev = "v${version}";
+    sha256 = "sha256-S2aUa25EnUfX6ShW6D+wHrsTLxTcCFcZ/uLE9EWGu4Q=";
```
```suggestion
    rev = "refs/tags/v${version}";
    hash = "sha256-S2aUa25EnUfX6ShW6D+wHrsTLxTcCFcZ/uLE9EWGu4Q=";
```

eclairevoyant reviewed (changes requested):
Other changes are okay, though note that we don't really need to order the input args, nor do we use `nixpkgs-fmt` in nixpkgs (the official formatter is `nixfmt-rfc-style`). If reformatting, that should be a separate commit ideally.

eclairevoyant commented on pkgs/top-level/all-packages.nix:
```diff
@@ -37516,11 +37516,6 @@ with pkgs;
-    lua = lua5_1;
-    inherit (darwin.apple_sdk.frameworks) Cocoa;
-  };
-
```
This should not be removed, rather the path should be changed to `../by-name/th/the-powder-toy/package.nix`

(Though it now uses lua 5.2, so should likely be pinned to that.)

LordGrimmauld replied:
Why is this? The line at https://github.com/NixOS/nixpkgs/blob/9863d8c74a959d73df0c495a55097f804f7b379d/pkgs/by-name/th/the-powder-toy/package.nix#L2 means the package behaves identical after the update as it did before (in regards to .override which as far as i understood is the motivation) even without listing in all-packages.nix, so why should it still be listed in all-packages.nix?

eclairevoyant replied:
lua versions aren't backwards-compatible
and in that case what's the point of having Cocoa as a function arg?

LordGrimmauld replied:
I have no idea why Cocoa is a function arg, it was there before i touched the module and figured getting rid of it would break the existing override behaviour. I'd be more than happy getting rid of it.

LordGrimmauld commented:
Addressed the feedback. Though i can't say i quite agree with it staying in all-packages.nix seeing as i tried keeping the override interface consistent, it is probably easier to just leave it in all-packages instead of arguing over it.
